src/components/Header.jsx

```diff
@@ -11,12 +11,11 @@ import { useRouter } from "next/navigation";
 export const Header = ({ session }) => {
   let UserGreetings;
   let Buttons;
-
-  if (!session) {
-    Buttons = <HeaderButtons type="Guest" />;
-  } else {
+  if (Object.keys(session).length > 0) {
```

Comment:
This is too technical. You can just check if session.token exists. or session.userData exists

```
if(session.token !== null)
```